service-base/src/test/java/org/eclipse/hono/service/management/credentials/SecretsTest.java

```diff
@@ -134,4 +143,30 @@ public void testDecodeGeneric() {
 
     }
 
+    /**
+     * Tests that secret dates are properly encoded and decoded.
+     * Both formats with timezone offset and without the offset are supported for encoding.
+     * For decoding the format without the offset is used.
+     */
+    @Test
+    public void testDateFormats() {
+
+        Json.mapper.registerModule(new JavaTimeModule());
+
+        final JsonObject json = new JsonObject()
+                .put(FIELD_SECRETS_COMMENT, "test")
+                .put(FIELD_SECRETS_NOT_BEFORE, "2017-05-01T14:00:00+01:00")
+                .put(FIELD_SECRETS_NOT_AFTER, "2037-06-01T14:00:00Z");
+
+
+        final PasswordSecret secret = json.mapTo(PasswordSecret.class);
+        final LocalDateTime before = LocalDateTime.of(2017, 05, 01, 14, 00, 00);
+        assertEquals(before.toInstant(ZoneOffset.of("+01:00")), secret.getNotBefore());
+
+        final JsonObject decodedSecret = JsonObject.mapFrom(secret);
+        assertEquals("2017-05-01T13:00:00Z", decodedSecret.getValue("not-before"));
```

Comment:
@dejanb Don't you think the expected value in this case should be: `2017-05-01T14:00:00+01:00`, exactly as was specified in the JSON object and not the value in `DateTimeFormatter.ISO_INSTANT` format?

Reply:
@Alfusainey the idea is that ISO_INSTANT is preferred format and that we just supporting defining dates in different formats for better usability.

Reply:
thanks 👍